packages/tests/esm/jest.config.ts

```diff
@@ -2,10 +2,13 @@ import type { Config } from "@jest/types";
 import { createDefaultEsmPreset } from "ts-jest";
 
 export default {
-  ...createDefaultEsmPreset({ isolatedModules: true }),
+  ...createDefaultEsmPreset(),
   testMatch: ["<rootDir>/src/**/*.test.(ts|tsx)"],
   globalSetup: "<rootDir>/src/setupTestEnv.ts",
   testEnvironment: "node",
+  moduleNameMapper: {
+    "^(\\.{1,2}/.*)\\.js$": "$1",
+  },
```

Comment:
Back in https://github.com/joist-orm/joist-orm/pull/1353, @stephenh mentioned that he had to use the `.ts` imports via `tsconfig.json` because Jest required it.

This is how you use ESM-style imports with ts-jest.

Reply:
I vaguely recall struggling this, i.e. getting jest to rewrite the .js-s back to .ts so it can load it off the file system, but then it would (to my recollection) break loading random things out of `node_modules` and other "wow I hate this" sort of weirdness.

Weird but makes sense solution that we just drop the js, don't add it back, and let jest reprobe "maybe that should have been js or ts or tsx or what not".